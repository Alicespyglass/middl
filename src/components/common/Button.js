import React from 'react';
import { Text, TouchableOpacity } from 'react-native';

const Button = ({ onPress, children }) => {
  const { buttonStyle, textStyle } = styles;

  return (
    <TouchableOpacity onPress={onPress} style={buttonStyle}>
      <Text style={textStyle}>
        {children}
      </Text>
    </TouchableOpacity>
  );
};

const styles = {
  textStyle: {
    alignSelf: 'center',
    // color: '#007aff',
    color: 'white',
    fontSize: 18,
    fontWeight: '600',
    paddingTop: 10,
    paddingBottom: 10
  },
  buttonStyle: {
    flex: 1,
    alignSelf: 'stretch',
    backgroundColor: 'rgba(83, 90, 146, 0.55)',
    borderRadius: 5,
    borderWidth: 1,
    borderColor: '#007aff',
    // borderColor: 'rgb(50, 50, 50)',
    marginLeft: 5,
    marginRight: 5
  }
};

export { Button };
