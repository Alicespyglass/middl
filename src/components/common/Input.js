import React from 'react';
import { TextInput, View, Text } from 'react-native';

const Input = ({ label, value, onChangeText, placeholder, secureTextEntry }) => {
  const { inputStyle, labelStyle, containerStyle } = styles;

  return (
    <View style={containerStyle}>
      <Text style={labelStyle}>{label}</Text>
      <TextInput
        secureTextEntry={secureTextEntry}
        placeholder={placeholder}
        placeholderTextColor='rgba(0, 0, 0, 0.3)'
        autoCorrect={false}
        style={inputStyle}
        value={value}
        onChangeText={onChangeText}
      />
    </View>
  );
};

const styles = {
  inputStyle: {
    color: '#222',
    paddingRight: 5,
    paddingLeft: 5,
    fontSize: 22,
    lineHeight: 23,
    flex: 2,
    fontFamily: 'System',
    // fontFamily: 'Heiti TC'
  },
  labelStyle: {
    color: '#333',
    fontSize: 22,
    paddingLeft: 20,
    flex: 1,
    fontWeight: 'bold',
    fontFamily: 'System',
    // fontFamily: 'Heiti TC'
  },
  containerStyle: {
    height: 40,
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  }
};

export { Input };
