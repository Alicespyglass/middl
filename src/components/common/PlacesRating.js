const PlacesRating = (ratArray) => {
  const ratArrayNoUndefined = ratArray.filter(function(n){ return n.rating !== undefined })
  const sortedArray = ratArrayNoUndefined.sort(function(a,b) {
    return b.rating - a.rating;
  });
  return { ratingsArray: sortedArray };
}

export { PlacesRating };
