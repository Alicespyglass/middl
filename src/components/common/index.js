export * from './Button';
export * from './Card';
export * from './CardSection';
export * from './Input';
