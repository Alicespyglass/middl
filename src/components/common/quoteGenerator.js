const quoteGenerator = {
  quotes: [
    '“If you’re afraid of butter, use cream.” -Julia Child',
    '“First we eat, then we do everything else.” -M.F.K. Fisher',
    '“Life is uncertain. Eat dessert first.” -Ernestine Ulmer',
    '“In wine there is wisdom, in beer there is strength, in water there is bacteria.” -David Auerbach',
    '“Wine and cheese are ageless companions, like aspirin and aches, or June and moon, or good people and noble ventures.” -M.F.K. Fisher',
    '“You don’t need a silver fork to eat good food.” -Paul Prudhomme',
    '“I have made a lot of mistakes falling in love, and regretted most of them, but never the potatoes that went with them.” -Nora Ephron',
    '“The only time to eat diet food is while you’re waiting for the steak to cook.” -Julia Child',
    '“I cook with wine. Sometimes I even add it to the food.” -W.C. Fields',
    '“We all eat, and it would be a sad waste of opportunity to eat badly.” -Anna Thomas',
    '“A balanced diet is a cookie in each hand.” -Barbara Johnson',
    '“People who love to eat are always the best people." -Julia Child',
    '“My doctor told me I had to stop throwing intimate dinners for four unless there are three other people.” -Orson Welles',
    '“The secret of success in life is to eat what you like and let the food fight it out inside.” -Mark Twain',
    '“It’s difficult to think anything but pleasant thoughts while eating a homegrown tomato.” -Lewis Grizzard',
    '“There is no sincerer love than the love of food.” -George Bernard Shaw',
    '“One cannot think well, love well, sleep well, if one has not dined well.” -Virginia Woolf',
    '“If you really want to make a friend, go to someone’s house and eat with him… the people who give you their food give you their heart.” -Cesar Chavez',
    '“My weaknesses have always been food and men — in that order.” - Dolly Parton',
    '“If more of us valued food and cheer and song above hoarded gold, it would be a merrier world.” -J.R.R. Tolkien',
    '“Cooking is like love. It should be entered into with abandon or not at all.” -Harriet van Horne',
    '“He was a bold man that first ate an oyster.” - Jonathan Swift',
    '“Food is symbolic of love when words are inadequate.” -Alan D. Wolfelt',
    '“Vegetables are a must on a diet. I suggest carrot cake, zucchini bread and pumpkin pie.” -Jim Davis',
    '“All happiness depends on a leisurely breakfast.” -John Gunther'
  ]
};

export { quoteGenerator };
