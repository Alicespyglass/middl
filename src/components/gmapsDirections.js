import React, { Component } from 'react';
import getDirections from 'react-native-google-maps-directions';

export default class gmapsDirections extends Component {

  handleGetDirections = () => {
    const data = {
       source: {
        latitude: -33.8356372,
        longitude: 18.6947617
      },
      destination: {
        latitude: -33.8600024,
        longitude: 18.697459
      },
      params: [
        {
          key: 'dirflg',
          value: 'w'
        }
      ]
    };

    getDirections(data);
  }

}
