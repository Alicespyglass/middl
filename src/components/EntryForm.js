import React, { Component } from 'react';
import { View, StyleSheet, Image, Text } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { Card, CardSection, Input, Button, quoteGenerator } from './common';

class EntryForm extends Component {
  constructor() {
    super();
    this.state = quoteGenerator;
  }

  componentWillMount() {
    this.setState({randomQuote: this.state.quotes[Math.floor(Math.random() * this.state.quotes.length)]});
  }


  render() {
    const goToResultsPageCafe = () => {
    Actions.results({ p1: this.state.personOneLocation,
      p2: this.state.personTwoLocation,
      placeType: 'Cafe'
      });
    };

    const goToResultsPageBar = () => {
    Actions.results({ p1: this.state.personOneLocation,
      p2: this.state.personTwoLocation,
      placeType: 'Bar'
      });
    };

    const goToResultsPageRestaurant = () => {
    Actions.results({ p1: this.state.personOneLocation,
      p2: this.state.personTwoLocation,
      placeType: 'Restaurant'
      });
    };

    return (
      <Image source={require('../assets/blurryTable.jpg')} style={styles.backgroundImage}>

        <View style={styles.container}>
          <Card>
            <CardSection style={styles.cardSection}>
              <Input
                label="You"
                placeholder="Where are you?"
                onChangeText={(personOneLocation) => this.setState({ personOneLocation })}
              />
            </CardSection>

            <CardSection style={styles.cardSection}>
                <Input
                  label="Them"
                  placeholder="Where's your friend?"
                  onChangeText={(personTwoLocation) => this.setState({ personTwoLocation })}
                />
            </CardSection>

            <CardSection>
              <Button
              accessibilityLabel='Click this button to find somewhere you and your friend can meet'
              onPress={goToResultsPageCafe}
              >
                Cafe
              </Button>
            </CardSection>

            <CardSection>
              <Button
              accessibilityLabel='Click this button to find somewhere you and your friend can meet'
              onPress={goToResultsPageBar}
              >
                Bar
              </Button>
            </CardSection>

            <CardSection>
              <Button
              accessibilityLabel='Click this button to find somewhere you and your friend can meet'
              onPress={goToResultsPageRestaurant}
              >
                Restaurant
              </Button>
            </CardSection>
          </Card>
        </View>

        <View style={styles.quotesView}>
          <Text style={styles.quotesText}>
            {this.state.randomQuote}
          </Text>
        </View>

      </Image>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  cardSection: {
    borderBottomWidth: 1,
  },
  backgroundImage: {
    flex: 1,
    alignSelf: 'stretch',
    width: null
  },
  quotesView: {
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    marginBottom: 60,
  },
  quotesText: {
    color: 'white',
    fontSize: 24,
    fontFamily: 'Bodoni 72',
    alignSelf: 'center',
    padding: 5
  }
});

export default EntryForm;
