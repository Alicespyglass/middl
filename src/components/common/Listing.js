import React from 'react';
import { Text, Linking, StyleSheet } from 'react-native';
import { Card, CardSection, Button } from '../common';
import { handleGetDirections, whatsappMessage, stars } from '../methods';

const Listing = ([name,
                  address,
                  rating,
                  placeType,
                  userLatitude,
                  userLongitude,
                  place1lat,
                  place1lng]) => {
  return (
    <Card>
      <CardSection style={styles.cardSection}>
        <Text style={styles.venueTitle}>
          {name}
        </Text>
      </CardSection>

      <CardSection style={styles.cardSection}>
        <Text style={styles.venueAddress}>
          {address}
        </Text>
      </CardSection>

      <CardSection style={styles.cardSection}>
        <Text>
          {stars(rating)}
        </Text>
      </CardSection>

      <CardSection style={styles.cardSection}>
        <Text style={styles.venueType}>
          {placeType}
        </Text>
      </CardSection>

      <CardSection style={styles.cardSection}>
        <Button
          onPress={() =>
            handleGetDirections(userLatitude,
              userLongitude,
              place1lat,
              place1lng)}
        >
          Get Directions
        </Button>

        <Button
          onPress={() =>
            Linking.openURL(whatsappMessage(name, address))}
        >
          Message friend
        </Button>
      </CardSection>
    </Card>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  venueTitle: {
    color: '#333',
    fontSize: 20,
    fontWeight: 'bold'
  },
  venueType: {
    color: '#333',
  },
  venueAddress: {
    color: '#333',
    fontStyle: 'italic'
  },
  cardSection: {
    justifyContent: 'center',
    alignItems: 'center'
  },
  backgroundImage: {
    flex: 1,
    alignSelf: 'stretch',
    width: null
  }
});

export { Listing };
