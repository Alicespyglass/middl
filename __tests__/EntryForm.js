import 'react-native';
import React from 'react';
import EntryForm from '../src/components/EntryForm';
import { shallow } from 'enzyme';
import { Card, CardSection, Input } from '../src/components/common';
import { View } from 'react-native';
import renderer from 'react-test-renderer';

describe('<EntryForm />', () => {
  it('renders one Card component', () => {
    const wrapper = shallow(<EntryForm />);
      expect(wrapper.find(Card).length).toBe(1);
  });

  it('renders three CardSection component', () => {
    const wrapper = shallow(<EntryForm />);
      expect(wrapper.find(CardSection).length).toBe(3);
  });

  it('renders two Input component', () => {
    const wrapper = shallow(<EntryForm />);
      expect(wrapper.find(Input).length).toBe(2);
  });

});

it('renders correctly', () => {
  const tree = renderer.create(
    <EntryForm />).toJSON();
    expect(tree).toMatchSnapshot();
});
