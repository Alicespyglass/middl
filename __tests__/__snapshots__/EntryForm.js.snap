// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#ddd",
        "borderRadius": 2,
        "borderWidth": 1,
        "elevation": 1,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 1,
            "borderColor": "#ddd",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 5,
            "position": "relative",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": 40,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            }
          }
        >
          You
        </Text>
        <TextInput
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="Where are you?"
          secureTextEntry={undefined}
          style={
            Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            }
          }
          value={undefined}
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 1,
            "borderColor": "#ddd",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 5,
            "position": "relative",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": 40,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            }
          }
        >
          Them
        </Text>
        <TextInput
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="Where's your friend?"
          secureTextEntry={undefined}
          style={
            Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            }
          }
          value={undefined}
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 1,
            "borderColor": "#ddd",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 5,
            "position": "relative",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#fff",
            "borderColor": "#007aff",
            "borderRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "marginLeft": 5,
            "marginRight": 5,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#007aff",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          GO
        </Text>
      </View>
    </View>
  </View>
</View>
`;
